pyrealm/splash/evap.py

```diff
@@ -0,0 +1,316 @@
+"""The evap submodule provides functions and classes to calculate evaporative fluxes."""
+
+from dataclasses import InitVar, dataclass, field
+from typing import Optional, Union
+
+import numpy as np
+
+from pyrealm.splash.const import kCw, kG, kL, kMa, kMv, kPo, kR, kTo, kw, pir
+from pyrealm.splash.solar import DailySolarFluxes
+from pyrealm.utilities import check_input_shapes
+
+
+@dataclass
+class DailyEvapFluxes:
+    """Calculate daily evaporative fluxes.
+
+    This class calculates daily evapotranspiration fluxes given temperature and
+    atmospheric pressure for daily observations and the calculated solar fluxes for
+    those observations. The :meth:`~pyrealm.splash.evap.DailyEvapFluxes.estimate_aet`
+    method can then be used to estimate actual evapotranspiration for observations,
+    given estimates of the evaporative supply rate.
+
+    Args:
+        solar: The daily solar fluxes for the observations
+        kWm: The maximum soil water capacity (mm)
+        tc: The air temperature of the observations (°C)
+        pa: The atmospheric pressure of the observations
+    """
+
+    solar: DailySolarFluxes
+    pa: InitVar[np.ndarray]
+    tc: InitVar[np.ndarray]
+    kWm: np.ndarray = np.array([150.0])
+
+    sat: np.ndarray = field(init=False)
+    """Slope of saturation vapour pressure temperature curve, Pa/K"""
+    lv: np.ndarray = field(init=False)
+    """Enthalpy of vaporization, J/kg"""
+    pw: np.ndarray = field(init=False)
+    """Density of water, kg/m^3"""
+    psy: np.ndarray = field(init=False)
+    """Psychrometric constant, Pa/K"""
+    econ: np.ndarray = field(init=False)
+    """Water-to-energy conversion factor"""
+    cond: np.ndarray = field(init=False)
+    """Daily condensation, mm"""
+    eet_d: np.ndarray = field(init=False)
+    """Daily EET, mm"""
+    pet_d: np.ndarray = field(init=False)
+    """Daily PET, mm"""
+    rx: np.ndarray = field(init=False)
+    """Variable substitute, (mm/hr)/(W/m^2)"""
+
+    def __post_init__(self, pa: np.ndarray, tc: np.ndarray) -> None:
+        """Calculate invariant components of evapotranspiration.
+
```

Comment:
I see what you are calculating here, this makes a lot of sense. I think the use of `evapotranspiration fluxes` and `evaporative fluxes` is inconsistent and a bit confusing. Maybe a way around it is listing the components and saying that they don't vary. something like ` this method calculates bla,bli, and blub which depend only on the input variables but are not affected by the model state of the previous time step.`